### PR TITLE
Fix issue 1319 - return correct code on bad / invalid requests

### PR DESCRIPTION
**Pull request by eladiw**
## Description
Instead of returning Forbidden. returning the correct code

## Related issues
Addresses [issue #1319].

### CODE REVIEW

**Comment by johnstairs** [comment]
@eladiw, the problem is that the [spec](https://www.hl7.org/fhir/search.html#errors) says we should return 403 in these cases. I agree it shouldn't be this way though.
>A HTTP status code of 403 signifies that the server refused to perform the search, while other 4xx and 5xx codes signify that some sort of error has occurred

@brendankowitz, what do you think?

File: src/Microsoft.Health.Fhir.Core/Exceptions/CustomerManagedKeyException.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public CustomerManagedKeyException(string message)
         {
             Issues.Add(new OperationOutcomeIssue(
                     OperationOutcomeConstants.IssueSeverity.Error,
-                    OperationOutcomeConstants.IssueType.Forbidden,
+                    OperationOutcomeConstants.IssueType.Invalid,
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Core/Exceptions/CustomerManagedKeyException.cs:17]: https://www.hl7.org/fhir/codesystem-issue-type.html#issue-type-no-store, thoughts?

**Comment by brendankowitz** [comment]
> @eladiw, the problem is that the [spec](https://www.hl7.org/fhir/search.html#errors) says we should return 403 in these cases. I agree it shouldn't be this way though.
> 
> > A HTTP status code of 403 signifies that the server refused to perform the search, while other 4xx and 5xx codes signify that some sort of error has occurred
> 
> @brendankowitz, what do you think?

If we also don't return the correct status code, we should also fix this, but it doesn't have to be in this PR since its just changing the issue types.

**Comment by brendankowitz** [comment]
I'll merge this as a partial fix for #1319. I've updating but bug with additional information about changing the status code.

**Review by brendankowitz — APPROVED** [review]

**Pull request merged**